src/reader.rs

```diff
@@ -5116,29 +5108,29 @@
             if search_term != zelf.pager.search_field_line.text() {
                 return; // Stale request.
             }
+            let history_len = zelf.history.size();
             let history_pager = zelf.history_pager.as_mut().unwrap();
-            history_pager.direction = direction;
-            match direction {
-                SearchDirection::Forward => {
-                    history_pager.can_go_backwards = true;
-                    if index == 0 || index <= result.final_index {
-                        return;
-                    }
-                    history_pager.history_index_start = result.final_index;
-                    history_pager.history_index_end = index;
-                }
-                SearchDirection::Backward => {
-                    history_pager.history_index_start = index;
-                    history_pager.history_index_end = result.final_index;
-                    history_pager.can_go_backwards = result.have_more_results;
-                }
-            };
-            zelf.pager.extra_progress_text = if result.have_more_results {
-                wgettext!("Search again for more results")
-            } else {
-                L!("")
+
+            if result.matched_commands.is_empty() && result.range != (0..history_len + 1) {
+                history_pager.start = history_len;
+                zelf.fill_history_pager(HistoryPagerInvocation::Advance, SearchDirection::Forward);
+                return;
             }
-            .to_owned();
+
+            assert!(result.range.start < result.range.end);
+            assert!(result.range.end <= history_len + 1);
+
+            *history_pager = result.range;
+            zelf.pager.extra_progress_text = if *history_pager != (0..history_len + 1) {
+                wgettext_fmt!(
+                    "Items %lu to %lu of %lu",
+                    history_pager.start + 1,
+                    history_pager.end - 1,
+                    history_len
+                )
```

Comment:
> Banner now should show index of first element of page (if there exists elements before it, otherwise 1) and index of first element of next page - 1 (or history.size() if there no next pages).

Looks good.